src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -150,41 +150,92 @@ public Type GetColumnSystemType(string columnName)
         /// valid foreign key definition. It is guaranteed at least one fk definition
         /// will be valid since the MetadataProvider.ValidateAllFkHaveBeenInferred.
         /// </summary>
+        /// <param name="relationshipName">Name of the relationship as represented in config file. The GraphQL
+        /// field referencing a nested entity has the same name as the relationship.</param>
         /// <param name="targetEntityName">Entity name as in config file for the related entity.</param>
         /// <param name="relatedSourceAlias">The alias assigned for the underlying source of this related entity.</param>
         /// <param name="subQuery">The subquery to which the join predicates are to be added.</param>
-        public void AddJoinPredicatesForRelatedEntity(
+        public void AddJoinPredicatesForRelatedEntityNEW(
+            string relationshipName,
+            string targetEntityName,
+            string relatedSourceAlias,
+            BaseSqlQueryStructure subQuery)
+        {
+            EntityRelationshipKey selfJoinKey = new(EntityName, relationshipName);
+            if (MetadataProvider.RelationshipToFkDefinitions.TryGetValue(key: selfJoinKey, out ForeignKeyDefinition? fkDef))
+            {
+                if (fkDef.ReferencedEntityRole is not RelationshipRole.Linking && fkDef.ReferencingEntityRole is not RelationshipRole.Linking)
```

Comment:
When is ReferencedEntityRole ever set to `Linking`?

Reply:
Set when a linking object is detected in the relationship within `private void AddForeignKeysForRelationships()`
https://github.com/Azure/data-api-builder/blob/5a09b5aafef7bbe44dcc74bd82a475cf7ea56be7/src/Core/Services/MetadataProviders/SqlMetadataProvider.cs#L771-L798

Reply:
Here the referencingentityrole is set to Linking right?

Reply:
yes